modules/simple_forms_api/app/services/simple_forms_api/metadata_validator.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
+  class MetadataValidator
+    def self.validate(metadata)
+      validate_first_name(metadata)
+        .then { |m| validate_last_name(m) }
+        .then { |m| validate_file_number(m) }
+        .then { |m| validate_zip_code(m) }
+        .then { |m| validate_source(m) }
+        .then { |m| validate_doc_type(m) }
+    end
+
+    def self.validate_first_name(metadata)
+      validate_presence_and_stringiness(metadata['veteranFirstName'], 'veteran first name')
+      metadata['veteranFirstName'] = metadata['veteranFirstName'][0..49]
+      metadata['veteranFirstName'] = metadata['veteranFirstName'].gsub(%r{[^a-zA-Z\-\/\s]}, '')
```

Comment:
again, this looks less like validation and more like normalization. is that intentional?

Reply:
Yes, you're right, this is beyond validation. It's intentional but maybe not great practice. This case could come up where the FE validations are out of sync with the LH validations (which is actually the case for us currently on 0845, as it turns out). I figured it's better to mangle their name on submission slightly than present the user with an error that won't make sense to them (ie, "Your name is invalid but I can't tell you why"). What do you think?

Reply:
I'm good with it, just wanted to be certain I understood the intention.